test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -319,3 +320,53 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 		require.True(tt, result, "system-probe and security-agent should communicate")
 	})
 }
+
+// CheckSystemProbeBehavior runs tests to check the agent behave correctly when system-probe is enabled
+func CheckSystemProbeBehavior(t *testing.T, client *TestClient) {
+	t.Run("enable system-probe and restarts", func(tt *testing.T) {
+		err := client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		err = client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Restart(client.Helper.GetServiceName())
+		require.NoError(tt, err, "datadog-agent should restart after CWS is enabled")
+	})
+
+	t.Run("system-probe is running", func(tt *testing.T) {
+		var err error
+		require.Eventually(tt, func() bool {
+			return AgentProcessIsRunning(client, "system-probe")
+		}, 1*time.Minute, 500*time.Millisecond, "system-probe should be running ", err)
+	})
+
+	t.Run("ebpf programs are unpacked and valid", func(tt *testing.T) {
+		ebpfPath := "/opt/datadog-agent/embedded/share/system-probe/ebpf"
+		output, err := client.Host.Execute(fmt.Sprintf("find %s -name '*.o'", ebpfPath))
+		require.NoError(tt, err)
+		files := strings.Split(strings.TrimSpace(output), "\n")
+		require.Greater(tt, len(files), 0, "ebpf object files should be present")
+
+		_, err = client.Host.Execute("command -v readelf")
```

Comment:
I have a question here: do we want to skip when the `readelf` command is not present, or is it better to install it? From what I've seen in the rest of the tests it looks like we aren't installing anything, that's why I skipped by default.

Reply:
Instead of readelf here can we debug/elf package to parse the sections? It would avoid having to install a new package and would make the test more robust IMO

Reply:
installing during the tests seems "risky", but if not available we could definitely add it to the build images

Reply:
Ah this is run on the remote host so we cannot read the file directly... Ignore my comment

Reply:
Yeah, this is on the remote. If possible to add to buildimages I think that would be great, it's only the binutils package and that might be useful for other tests too.

Reply:
Ah and this isn't build image related :sweat_smile: , but we could probably add it to the E2E images